docker/worker-base/Dockerfile

```diff
@@ -35,15 +35,21 @@ RUN curl -sS https://www.python.org/ftp/python/${PYTHON_VERSION}/Python-${PYTHON
     make -j && \
     make install && \
     rm -rf /tmp/Python-${PYTHON_VERSION}
-RUN pip3 install pipenv==2023.12.1
+
+# Setup Poetry in its own virtual environment.
+# So when poetry changes the system dependencies, it doesn't uninstall itself.
```

Comment:
This is pretty surprising. Do we know why poetry does this?

Looking at https://python-poetry.org/docs/cli/#options-2, it's not clear that this is the default behaviour.  `--sync` or `--remove-untracked` doesn't seem like default behaviour?

Reply:
The comment itself is probably not fully accurate, yes uninstalling itself completely is unlikely, but we could have dependencies that overlap with poetry's own dependencies, therefore changing the version unexpectedly.

This is the biggest/first warning on the installation instructions: https://python-poetry.org/docs/#installation

> Poetry should always be installed in a dedicated virtual environment to isolate it from the rest of your system. It should in no case be installed in the environment of the project that is to be managed by Poetry. This ensures that Poetry’s own dependencies will not be accidentally upgraded or uninstalled. (Each of the following installation methods ensures that Poetry is installed into an isolated environment.) In addition, the isolated virtual environment in which poetry is installed should not be activated for running poetry commands.

Reply:
Got it, thanks! Could you add this link and detail as a comment here?

Reply:
Done